include/emds/emds.h

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file emds.h
+ *
+ *  @defgroup emds Emergency Data Storage Library
+ *  @{
+ *  @brief Emergency Data Storage API
+ */
+
+#ifndef EMDS_H__
+#define EMDS_H__
+
+#include <stddef.h>
+#include <sys/types.h>
+#include <sys/util.h>
+#include <sys/slist.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @struct emds_entry
+ *
+ * Information about entries used to store data in the emergency data storage.
+ */
+struct emds_entry {
+	/** Unique ID for each static and dynamic entry. */
+	uint16_t id;
+	/** Pointer to data that will be stored. */
+	uint8_t *data;
+	/** Length of data that will be stored. */
+	size_t len;
+};
+
+/**
+ * @struct emds_dynamic_entry
+ *
+ * Entries with a node element, used for dynamic entries. These are registered
+ * using a call to @ref emds_entry_add.
+ */
+struct emds_dynamic_entry {
+	struct emds_entry entry;
+	sys_snode_t node;
+};
+
+/**
+ * @brief Define a static entry for emergency data storage items.
+ *
+ * @param name The entry name.
+ * @param _id Unique ID for the entry. This value and not an overlap with any
+ *            other value.
+ * @param _data Data pointer to be stored at emergency data store.
+ * @param _len Length of data to be stored at emergency data store.
+ *
+ * This creates a variable _name prepended by emds_.
+ */
+#define EMDS_STATIC_ENTRY_DEFINE(_name, _id, _data, _len)                      \
```

Comment:
minor: alignment of multiline macro

Reply:
This is aligned in the code.  This is mismatch in how it shows in github.

Reply:
You have extra spaces instead of tabs.

Reply:
I have checked there is tabs where the lines are starting, and spaces after the code until \. This is done in most of the other places this is used.